hack/e2e-suite/guestbook.sh

```diff
@@ -33,25 +33,58 @@ export KUBECTL KUBE_CONFIG_FILE
 source "${KUBE_ROOT}/cluster/kube-env.sh"
 source "${KUBE_VERSION_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
 
+function wait_for_running() {
+  echo "Waiting for pods to come up."
+  frontends=$(${KUBECTL} get pods -l name=frontend -o template '--template={{range.items}}{{.id}} {{end}}')
+  master=$(${KUBECTL} get pods -l name=redis-master -o template '--template={{range.items}}{{.id}} {{end}}')
+  slaves=$(${KUBECTL} get pods -l name=redisslave -o template '--template={{range.items}}{{.id}} {{end}}')
+  pods=$(echo $frontends $master $slaves)
+
+  all_running=0
+  for i in $(seq 1 30); do
+    sleep 10
+    all_running=1
+    for pod in $pods; do
+      status=$(${KUBECTL} get pods $pod -o template '--template={{.currentState.status}}') || true
+      if [[ "$status" == "Pending" ]]; then
```

Comment:
Just because pods are in the running state doesn't mean the docker pulls have finished and that the application is ready to serve. Would it be more reliable to do a loop around a get from the frontend service which might be a better signal that the e2e test should proceed.